engine/src/main/java/org/terasology/utilities/procedural/SubSampledNoise.java

```diff
@@ -121,6 +173,22 @@ public float noise(float x, float y) {
         return fullData;
     }
 
+
+    private float[] getKeyValues(BlockAreac fullRegion) {
+        int xDim = fullRegion.getSizeX() / sampleRate + 1;
+        int yDim = fullRegion.getSizeY() / sampleRate + 1;
+        float[] fullData = new float[xDim * yDim];
+        for (int y = 0; y < yDim; y++) {
+            for (int x = 0; x < xDim; x++) {
+                int actualX = x * sampleRate + fullRegion.minX();
+                int actualY = y * sampleRate + fullRegion.minY();
+                fullData[x + y * xDim] = source.noise(zoom.x * actualX, zoom.y * actualY);
+            }
+        }
+
+        return fullData;
+    }
+
     private Rect2i determineRequiredRegion(Rect2i region) {
```

Comment:
Is this method still used?

Reply:
its an internal method so its used within the class. Am I misunderstanding the question?

Reply:
Hm, I think I'm confused by the duplicate code. Instead of keeping the old implementation on the TeraMath types we could completely remove it and only migrate the public entry point for this class, which is the `public float[] noise(...)` function.